### ISSUE TITLE
How to get the OpenSeadragon.MouseTracker events when using annotorious plugin

### ISSUE THREAD

**Issue by vivsamcom** [openseadragon/openseadragon#2289]
Hi,

I have included Annotorious Toolbar,Annotorious Selector Pack and it works as expected.
I have crop functionality also wherein I am using OpenSeadragon.MouseTracker,OpenSeadragon.Rect to get the croped area.
With the inclusion of the Annotorious plugins now I am not able to select the data using OpenSeadragon functions.
I have tried to use anno.setDrawingEnabled(true); on click of crop button. However it is not working. Am I missing somethging here?

Mine openseadragon version is 2.4.2 and Annotorious version 2.6.0.

Asked the question in annotorious-openseadragon forum as well.
https://github.com/recogito/annotorious-openseadragon/issues/158 

Thanks

**Comment by iangilman** [comment]
@msalsbery Any thoughts? 

@vivsamcom I don't know enough about that plugin to know what to suggest. It's good you posted there as well; hopefully you'll hear back!

**Comment by msalsbery** [comment]
@vivsamcom  @iangilman 

The OpenSeadragon version is pretty old. There's been a lot of enhancements to make event handling easier for things like annotations but I don't know what OSD version the plugin supports or how the plugin handles events

**Comment by vivsamcom** [comment]
@iangilman,

Is there a way I can draw line or free form using the OpenSeadragon  functions  just like [OpenSeadragon](http://openseadragon.github.io/docs/OpenSeadragon.html).Rect  without using any other plugin.

**Comment by iangilman** [comment]
@vivsamcom I suppose you could. You'll need to figure out how you want to display that line. OSD does its drawing on canvas, so you can use canvas's drawing commands after each OSD frame, if you want to. As for tracking the user drawing the line, you'll need to use OSD's drag events and also turn off the built-in panning. 

Those are a lot of the things the various annotation and overlay plugins do, so you could look at their code for inspiration. 

I assume you want to do this because the existing plugins don't do what you want? If I was in that situation, I would probably still use one of overlay plugins, like SVG or canvas, just to make the drawing easier.

**Comment by overide** [comment]
Following up on this, I'm also facing the same issue. If I don't initialize the Annotorious, MouseTracker fires the click event properly. But I cannot disable the Annotorious in my project, I somehow want both to work together. 

@iangilman Is there any other way I can attach the click event on overlay directly? I tried adding a normal javascript event but it's not working without mouseTracker.

**Comment by iangilman** [comment]
@msalsbery Any thoughts from the event handling side?

@rsimon Any thoughts from the Annotorious side?

@overide Have you tried filing an issue on https://github.com/annotorious/annotorious-openseadragon?

**Comment by overide** [comment]
Issue was already opened by @vivsamcom in Annotorious-OSD repository but it was never addressed.

**Comment by msalsbery** [comment]
> @msalsbery Any thoughts from the event handling side?

Yes!

@overide 

[This codepen](https://codepen.io/msalsbery/pen/dyOeXqO) shows ways you can control MouseTracker on overlay elements. The MouseTracker for overlay1 demonstrates some possible click handling

**Comment by rsimon** [comment]
Event interplay was always notoriously tricky. Annotorious uses a MouseTracker and somehow that seems to stop event propagation.

By now, I think you might be better off trying the upcoming v3 of AnnotoriousOSD from https://github.com/annotorious/annotorious. Documentation is still almost non-existent. But several large-ish projects are using the package, so technically it's stable. I scheduled some time to work on docs in the latter half of August.

**Comment by msalsbery** [comment]
@rsimon pointer event handling has always been notoriously tricky 😅 but MouseTracker shouldn't be stopping any propagation except when creating gestures from primitive events where the pointer is captured. It should be all overridable at this point, however.

If you're having any issues I would be more than happy to help and/or patch OSD. All the work I've done on MouseTracker was specifically for working with annotation overlays while keeping backward OSD compatibility (the tricky part) so I always have a vested interest 😉

**Comment by rsimon** [comment]
@msalsbery I didn't want to imply that that's OSD's fault. Sorry if that came across the wrong way. The tricky part was always getting the event handling right in Annotorious as such. Also, I'm pretty sure that the culprit for this particular problem is this line here in Annotorious:
https://github.com/annotorious/annotorious-openseadragon/blob/main/src/OSDAnnotationLayer.js#L183

TBH I forgot why this was needed in the first place, but had something to do with different modes of starting annotations with a single click vs. a pointer drag.

In the end things got too messy an Annotorious - another reason I started the rewrite. The rewrite is still a bit convoluted, because I have some code that needs to work both with OSD as well as within a "normal" SVG (overlaid on an image). But I think I managed to get it right (or at least much better) this time.

**Comment by msalsbery** [comment]
@rsimon haha I didn't mean to imply it's NOT OSDs fault either. 

There's a lot going on in MouseTracker so there's so many ways it can go wrong, or just be harder to use than pointer event handling already is.

I'm just always looking for use cases and places OSD can improve, whether it's bug fixes or enhancements. 

Cheers 😄

**Comment by rsimon** [comment]
@iangilman that's right. It's not ready for prime time yet. I've reserved some time in August to finalize stuff. Hopefully I'll be able to make the official release soon.

**Comment by overide** [comment]
I'm using following at the moment - 

annotorious-openseadragon - v2.7.8
openseadragon: 3.1.0
